engine/lite/kafka/runtime/src/main/scala/pl/touk/nussknacker/engine/lite/kafka/KafkaProducerRecordsHandler.scala

```diff
@@ -0,0 +1,125 @@
+package pl.touk.nussknacker.engine.lite.kafka
+
+import com.github.ghik.silencer.silent
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.kafka.clients.consumer.{ConsumerConfig, ConsumerRecords, KafkaConsumer, OffsetAndMetadata}
+import org.apache.kafka.clients.producer.{KafkaProducer, ProducerRecord, RecordMetadata}
+import org.apache.kafka.common.{IsolationLevel, Metric, MetricName, TopicPartition}
+import pl.touk.nussknacker.engine.kafka.KafkaUtils
+import pl.touk.nussknacker.engine.lite.kafka.KafkaTransactionalScenarioInterpreter.KafkaInterpreterConfig
+import pl.touk.nussknacker.engine.util.Implicits._
+
+import java.util
+import java.util.{Properties, UUID}
+import scala.concurrent.Future
+import scala.jdk.CollectionConverters._
+
+object KafkaProducerRecordsHandler extends LazyLogging {
+
+  def apply(engineConfig: KafkaInterpreterConfig, producerProperties: Properties, consumerGroupId: String): KafkaProducerRecordsHandler = {
+    val shouldProcessRecordsInTransactions = engineConfig.kafkaTransactionsEnabled.getOrElse {
+      // Event hubs doesn't support transactional producers for now - it causes error on KafkaProducer.initTransactions()
+      // see https://github.com/Azure/azure-event-hubs-for-kafka/issues/209#issuecomment-1412041269
+      !engineConfig.kafka.kafkaBootstrapServers.exists(_.contains(".servicebus.windows.net"))
```

Comment:
Can we extract helper method with logic for desiding wheter is azure or not? 
I believe that now, considering this PR and the other with Azure support, it's in two places.

Reply:
For now, I extracted private method and left TODO